### PR TITLE
Feature/add progress bar indt

### PR DESCRIPTION
**Pull request by ektasawant**
Fixes PXBLUE-2273 .

#### Changes proposed in this Pull Request:
- Add Progress bar indeterminate

#### 
image

#### To Test:
- Local Test URL http://localhost:3000/progress-bar-indeterminate

### CODE REVIEW

File: src/pages/loading-waiting-states/progress-bar-indeterminate/index.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+const useStyles = makeStyles((theme: Theme) => ({
+    container: {
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        height: `calc(100vh - ${theme.spacing(8)}px)`,
+    },
+    appbarRoot: {
+        padding: 0,
+    },
```
Comment (joebochill) [inline src/pages/loading-waiting-states/progress-bar-indeterminate/index.tsx:20]: I don't think this class is necessary...the AppBar should already have no padding by default.

File: src/pages/loading-waiting-states/progress-bar-indeterminate/index.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    toolbarGutters: {
+        padding: '0 16px',
```
Comment (joebochill) [inline src/pages/loading-waiting-states/progress-bar-indeterminate/index.tsx:22]:
```suggestion
        padding: `0px ${theme.spacing(2)}px`,
```

File: src/pages/loading-waiting-states/progress-bar-indeterminate/index.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    },
+    LinearProgressBarIndeterminate: {
+        position: 'absolute',
+        bottom: 0,
+        marginLeft: '-16px',
```
Comment (joebochill) [inline src/pages/loading-waiting-states/progress-bar-indeterminate/index.tsx:27]: use `theme.spacing`

**Review by joebochill — COMMENTED** [review]
I think we need to fix the routing...looks like 4 different routes are active on the `/` route:
image

**Comment by joebochill** [comment]
Looks good otherwise

**Review by emclaug2 — APPROVED** [review]
Looks good to me, but one I noticed (not caused by your PR) is that whenever I select an item from the navigation menu, all nested nav items become expanded.  Should this marked as an issue @joebochill  @huayunh ?

**Comment by daileytj** [comment]
> Looks good to me, but one I noticed (not caused by your PR) is that whenever I select an item from the navigation menu, all nested nav items become expanded.  Should this marked as an issue @joebochill  @huayunh ? 

I haven't investigated it thoroughly, but I'm having a similar issue with the react showcase demo. This may be an issue with the drawer component.

**Comment by emclaug2** [comment]
> > Looks good to me, but one I noticed (not caused by your PR) is that whenever I select an item from the navigation menu, all nested nav items become expanded.  Should this marked as an issue @joebochill  @huayunh ?
> 
> I haven't investigated it thoroughly, but I'm having a similar issue with the react showcase demo. This may be an issue with the drawer component.

I can't recreate this in our storybook full config story, so I'm hoping this isn't a component-level issue.

**Comment by joebochill** [comment]
The issue with the expanding drawer items is not an issue with the component, but an issue with the route configuration in the particular project. There are multiple DrawerNavItems that have the same ID, so when you select a nested item and it calls the 'updateParent' function, all of the top level items match the parent id and so they get expanded.

This is caused in App.tsx on line 57, where we set the itemID...all of the top level items are given the `''` id:
image


This should be updated to be a unique value for each navigation item. A quick way to do this is to update the `createRoute` function to take the `key` of the item as well as it's data:
image


And then set the itemID to be `page.route || itemKey` instead of `page.route || ''`

File: src/App.tsx [hunk 4]
```diff
@@ -77,7 +77,7 @@ export const App: React.FC = () => {
                 },
             }}
             variant={isMobile ? 'temporary' : 'permanent'}
-            activeItem={selected}
+            activeItem={selected !== '' ? selected : undefined}
```
Comment (joebochill) [inline src/App.tsx:80]: I think you can probably undo this now, since none of the routes will have `''` as their id anymore.

**Review by joebochill — APPROVED** [review]

**Pull request merged**